### DISCUSSION TITLE
Switch to using linuxstatic for CLI builds

### DISCUSSION THREAD

**Discussion by mveytsman** [tailwindlabs/tailwindcss#6785] [Ideas]
Checking in here before I submit a drive-by PR :)

The current linux binaries are dynamically linked, which means they don't work out of the box (and give a confusing error) on minimal linux systems such as Alpine. You can see examples at https://github.com/tailwindlabs/tailwindcss/issues/6690 and https://github.com/phoenixframework/tailwind/issues/7 . The user would have to know that "not found" when running a binary indicates that a dynamically linked library is missing and know to install both `build-base` and `gcompat`.

We can't fix the confusing errors, but we can ship statically linked binaries.

Addressing some possible concerns:

**File size**
When I built it on my machine the static builds actually took up *less space* than dynamic

| Arch   | Dynamic | Static |
| -- | ------------- | ------------- |
| x64 | 41mb | 39mb |  
| arm64 | 43mb | 36mb |
 
**Native bindings**
The [pkg README](https://github.com/vercel/pkg#native-addons) indicates that "native addons" are not available when statically packaging -- I don't believe tailwindcss uses them and the linux static binary worked correctly when I tested it.

I think testing the linux binary as part of CI would be a good addition - I suggested doing so here: https://github.com/tailwindlabs/tailwindcss/discussions/6784

---

The change itself is pretty trivial (https://github.com/mveytsman/tailwindcss/commit/4b5cc157793bfce3dfb46c9834e70a63ccb80810) and I'm happy to submit this as a PR if the team thinks it's good :)

**Comment by eboskma** [comment]
I'm also having issues on NixOS. The usual ELF patching is not working, the CLI then spits out this error:

```
Pkg: Error reading from file.
```

A static build works fine.

**Comment by mveytsman** [comment]
@eboskma good to know! It looks like there's a NixOS package but it's installing via node? https://github.com/NixOS/nixpkgs/issues/149333

**Comment by mveytsman** [comment]
I made a PR for this in https://github.com/tailwindlabs/tailwindcss/pull/6914

**Comment by eboskma** [comment]
Ah, I missed that one, thanks for pointing it out. 

I was using the tailwind cli through the [elixir package](https://github.com/phoenixframework/tailwind) in an Elixir Phoenix project, and by default that downloads the executable from GitHub, which does not work on NixOS.

**Comment by mveytsman** [comment]
I'm here to make that package work for me as well 😁
